scripts/import-update-key

```diff
@@ -10,11 +10,35 @@ if [ -z "$keyfile" ]; then
   exit 1
 fi
 
-keyhash=$(gpg $gpgopts --with-fingerprint $keyfile | sed -ne 's#^pub  [^ ]\+/\([^ ]\+\).*#\1#p')
-if [ ${#keyhash} -ne 8 ]; then
+runver=$(gpg --version |grep -oP "gpg\s\(GnuPG\)\s\K.*") # runtime version
+tarver="2.4.0" # target version
+
+# version little or equal
+verlte() {
+  printf '%s\n' "$1" "$2" | sort -C -V
+}
+# version little than
+verlt() {
+    [ "$1" = "$2" ] && return 1 || verlte $1 $2
+}
+# Whether using legacy gpg
+legacygpg(){
+    verlt $runver $tarver
+}
+
+if legacygpg; then
+  key=$(gpg $gpgopts --with-fingerprint $keyfile | sed -ne 's#^pub  [^ ]\+/\([^ ]\+\).*#\1#p')
+else
+  key=$(gpg $gpgopts --list-packets $keyfile | grep -Po 'keyid:\s\K.*')
+fi
+
+keylen=$(legacygpg && echo "8" || echo "16")
+if [ ${#key} -ne $keylen ]; then
   echo "Invalid keyfile" >&2
   exit 2
 fi
+
 cp -p $keyfile /etc/pki/rpm-gpg
+
 gpg $gpgopts --import $keyfile
-echo -e "trust\n5\ny\n" | gpg $gpgopts --command-fd 0 --edit-key $keyhash
+echo -e "trust\n5\ny\n" | gpg $gpgopts --command-fd 0 --edit-key $key
```

Comment:
There is a `--trusted-key` flag that might be settable from `gpg.conf` (https://www.gnupg.org/(en)/documentation/manuals/gnupg-devel/GPG-Configuration-Options.html). The flag is at least present on the old version too, so it might be possible to implement this by writing to a configuration file instead of driving gpg with interactive commands.

Reply:
Manage the configure file is more error-prone than the comandline.